pkg/api/mlflow/service/artifact/service.go

```diff
@@ -6,29 +6,42 @@ import (
 	"github.com/G-Research/fasttrackml/pkg/api/mlflow/api"
 	"github.com/G-Research/fasttrackml/pkg/api/mlflow/api/request"
 	"github.com/G-Research/fasttrackml/pkg/api/mlflow/dao/repositories"
+	"github.com/G-Research/fasttrackml/pkg/api/mlflow/service/artifact/storage"
 )
 
 // Service provides service layer to work with `artifact` business logic.
 type Service struct {
-	runRepository repositories.RunRepositoryProvider
+	runRepository   repositories.RunRepositoryProvider
+	artifactStorage storage.Provider
 }
 
 // NewService creates new Service instance.
-func NewService(runRepository repositories.RunRepositoryProvider) *Service {
+func NewService(artifactStorage storage.Provider, runRepository repositories.RunRepositoryProvider) *Service {
 	return &Service{
-		runRepository: runRepository,
+		runRepository:   runRepository,
+		artifactStorage: artifactStorage,
 	}
 }
 
-func (s Service) ListArtifacts(ctx context.Context, req *request.ListArtifactsRequest) error {
+// ListArtifacts handles business logic of `GET /artifacts/list` endpoint.
+func (s Service) ListArtifacts(
```

Comment:
I kind of think the service layers should take more "neutral" argument types than api http request struct -- ie, controller should invoke service with string, int, etc, instead of passing on the http request -- this would aid in reuse across mflow and aim.